synthesizer/snark/src/proving_key/mod.rs

```diff
@@ -91,3 +91,21 @@ impl<N: Network> Deref for ProvingKey<N> {
         &self.proving_key
     }
 }
+
+impl<N: Network> PartialEq for ProvingKey<N> {
+    fn eq(&self, other: &Self) -> bool {
+        self.deref() == other.deref()
+    }
+}
```

Comment:
this is how `PartialEq` is implemented for `Arc<T>`; did the compiler complain when you tried to just derive it?